frontend/src/pages/schedules/shared.tsx

```diff
@@ -0,0 +1,29 @@
+import { NextPage } from "next";
+import React, { useEffect } from "react";
+import { useRouter } from "next/router";
+import Loading from "../../components/Loading";
+import { getCourseIds } from "../../app/utils";
+import { useAppDispatch } from "../../app/hooks";
+import { userSlice } from "../../app/user";
+
+const SharedSchedulePage: NextPage = () => {
+  const router = useRouter();
+  const dispatch = useAppDispatch();
+  const coursesString = router.query.courses as string;
+
+  useEffect(() => {
+    if (router.isReady) {
+      const courseIDs = getCourseIds(coursesString);
+      dispatch(
+        userSlice.actions.createSharedSchedule({
+          courses: courseIDs,
+        })
+      );
+      router.push("/schedules");
+    }
+  }, [router.isReady]);
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has missing dependencies: 'coursesString', 'dispatch', and 'router'. Either include them or remove the dependency array.](https://app.codacy.com/gh/ScottyLabs/course-tool/pullRequest?prid=9667658)
```suggestion
  }, [coursesString, dispatch, router, router.isReady]);
```